lib/fastlane/plugin/firebase_app_distribution/helper/firebase_app_distribution_auth_client.rb

```diff
@@ -23,43 +25,45 @@ module FirebaseAppDistributionAuthClient
       #   debug - Whether to enable debug-level logging
       #
       # env variables
-      #   GOOGLE_APPLICATION_CREDENTIALS - see google_service_path
       #   FIREBASE_TOKEN - see firebase_cli_token
       #
       # Crashes if given invalid or missing credentials
-      def fetch_auth_token(google_service_path, firebase_cli_token, debug = false)
+      def get_authorization(google_service_path, firebase_cli_token, debug = false)
         if !google_service_path.nil? && !google_service_path.empty?
           UI.message("🔐 Authenticating with --service_credentials_file path parameter: #{google_service_path}")
-          token = service_account(google_service_path, debug)
+          service_account(google_service_path, debug)
         elsif !firebase_cli_token.nil? && !firebase_cli_token.empty?
           UI.message("🔐 Authenticating with --firebase_cli_token parameter")
-          token = firebase_token(firebase_cli_token, debug)
+          firebase_token(firebase_cli_token, debug)
         elsif !ENV["FIREBASE_TOKEN"].nil? && !ENV["FIREBASE_TOKEN"].empty?
           UI.message("🔐 Authenticating with FIREBASE_TOKEN environment variable")
-          token = firebase_token(ENV["FIREBASE_TOKEN"], debug)
-        elsif !ENV["GOOGLE_APPLICATION_CREDENTIALS"].nil? && !ENV["GOOGLE_APPLICATION_CREDENTIALS"].empty?
-          UI.message("🔐 Authenticating with GOOGLE_APPLICATION_CREDENTIALS environment variable: #{ENV['GOOGLE_APPLICATION_CREDENTIALS']}")
-          token = service_account(ENV["GOOGLE_APPLICATION_CREDENTIALS"], debug)
+          firebase_token(ENV["FIREBASE_TOKEN"], debug)
+        elsif !application_default_creds.nil?
+          UI.message("🔐 Authenticating with Application Default Credentials")
+          application_default_creds
         elsif (refresh_token = refresh_token_from_firebase_tools)
-          UI.message("🔐 No authentication method specified. Using cached Firebase CLI credentials.")
-          token = firebase_token(refresh_token, debug)
+          UI.message("🔐 No authentication method found. Using cached Firebase CLI credentials.")
+          firebase_token(refresh_token, debug)
         else
           UI.user_error!(ErrorMessage::MISSING_CREDENTIALS)
+          nil
         end
-        token
       end
 
       private
 
+      def application_default_creds
+        Google::Auth.get_application_default([SCOPE])
+      rescue
+        nil
+      end
+
       def refresh_token_from_firebase_tools
         config_path = format_config_path
         if File.exist?(config_path)
           begin
             firebase_tools_tokens = JSON.parse(File.read(config_path))['tokens']
-            if firebase_tools_tokens.nil?
-              UI.user_error!(ErrorMessage::EMPTY_TOKENS_FIELD)
```

Comment:
why is this removed?

Reply:
I might move this to a separate PR. I realized that this isn't actually an exceptional case. If they log out of the Firebase CLI (at least the version I'm using), that file will still exist it just won't have a `tokens` field. In that case we just want to return `nil` and continue.